docs/manual/guides/deployer.en.md

```diff
@@ -0,0 +1,278 @@
+---
+title: "Deploy with Deployer"
+description: "Learn how to deploy your project with Deployer."
+aliases:
+    - /en/guides/deployer/
+weight: 80
+tags: 
+    - "Deployer"
+    - "Deployment"
+---
+
+{{% notice note %}}
+The Deployer recipe is part of Deployer 7 and is intended to work for Contao 4.13 and greater.
+
+## Install Deployer
+
+If not done yet, install Deployer as described here: [https://deployer.org/docs/][1]
+
+Verify that you are running Deployer in the minimum version _7.0.0-rc.5_ by running `dep --version`.
+
+Once done, you can create a `deploy.php` file in your project:
+
+```bash
+touch deploy.php
+```
+
+## Write deploy.php file
+
+There are two ways to deploy your project to the remote site. The default approach to deploy files in Deployer is to
+check out the Git repository with the current project on the remote server.
+
+### Option 1: Deploy with Repository
+
+To get started with Deployer, use the following file contents for the `deploy.php` in your project root:
+
+```php
+<?php // /deploy.php
+
+namespace Deployer;
+
+import('recipe/contao.php');
+
+host('example.org')
+    ->set('remote_user', 'foobar')
+    ->set('deploy_path', '/var/www/{{remote_user}}/html/{{hostname}}')
+    ->set('bin/php', 'php8.1')
+    ->set('bin/composer', '{{bin/php}} /var/www/{{remote_user}}/composer.phar')
+;
+
+set('repository', 'git@github.com:acme/example.org.git');
+
+set('keep_releases', 10);
+
+after('deploy:failed', 'deploy:unlock');
+```
+
+Make sure to adjust the host configuration ([Documentation][2]) and repository URL as required.
+
+The deployment with Git repo has some downsides, though. First, you always need to have your local files committed
+and pushed. Second, your remote site needs to have read-access on Git repository, which requires storing the HTTPS
+credentials or configuring SSH. Therefore, another favored approach is to use `rsync` instead of Git to deploy the
+project.
+
+### Option 2: Deploy with rsync
+
+To use `rsync` instead of a Git checkout, we need to use the rsync recipe and add further configuration to the
+`deploy.php` file.
+
+First we create a new recipe solely for the rsync functionality:
+
+```bash
+touch deploy-rsync.php
+```
+
+```php
+<?php // /deploy-rsync.php
+
+namespace Deployer;
+
+import('contrib/rsync.php');
```

Comment:
I never use this and `exclude` it's way too cumbersome. Why not just use the built-in `upload()` and list what you actually want to upload? It's better to say what you want than what you don't want anyway :)

Reply:
From my experience, to `include` is harder to configure, and much more project-specific, than to `exclude`.

**Example 1:**

Exclude the parameters.yml is as easy as excluding "config/parameters.yml" while using `upload()` would require defining each file, i.e., `config/services.yml`, `config/config_prod.yml`, `config/config.yml`, etc.

**Example 2:**

Taking for example my [theme-framework](https://github.com/richardhj/contao-theme-framework#1-create-a-theme-folder), where you shall exclude the "assets" folder from the theme directory when deploying, configuring rsync is as easy as excluding `'/themes/*/assets'`. While using `upload()` would require to upload each but one path: `/themes/foobar/public`, `/themes/foobar/templates`, `/themes/foobar/theme.yaml`.

In the end, to `include` requires much more know-how and customization than to `exclude`.

Reply:
Then I'll probably never use this recipe 🙈

Reply:
We can also introduce an "Option 3" :-) 

1. Git
2. rsync
3. scp

Reply:
Is there an option to use accordeons for code snippets in Hugo MD?

Reply:
```
{{% expand "Your text" %}}
…
{{% /expand %}}
```

Reply:
I just realized that `upload()` is using rsync and not scp in fact. https://github.com/deployphp/deployer/blob/c3d236a81f35e26592c8fa51b5f81639cd52db7c/src/functions.php#L578

Reply:
@Toflar I just added a section about using upload() in a72d2399ea5486561102a83f91c82e2126a67429. Can you add your thoughts? :-)